src/lib/FreeC/Backend/Agda/Syntax.hs

```diff
@@ -3,7 +3,74 @@
 
 module FreeC.Backend.Agda.Syntax
   ( module Agda.Syntax.Concrete
+  , module Agda.Syntax.Common
+  , module Agda.Syntax.Position
+    -- * Identifiers
+  , name
+  , qname
+  , qname'
+    -- * Imports
+  , simpleImport
+    -- * Expressions
+  , intLiteral
+  , lambda
   )
 where
 
 import           Agda.Syntax.Concrete
+import           Agda.Syntax.Position
+import           Agda.Syntax.Common
+
+import qualified Agda.Syntax.Common            as Agda
+import qualified Agda.Syntax.Concrete          as Agda
+import qualified Agda.Syntax.Literal           as Agda
+import qualified Agda.Syntax.Position          as Agda
```

Comment:
Importing the modules both qualified and unqualified has the unfortunate side effect that the `Agda` prefix can be forgotten without GHC complaining (for example, I just noticed that `QName` in `qname` is not qualified). Since it could be confusing if we mix `QName`  and `Agda.QName` but only unqualified imports can be reexported, the best option is probably to remove the qualified imports and not qualify identifiers from the Agda library in this module. I'll followup with the corresponding change to `FreeC.Backend.Coq.Syntax`.

Reply:
This sounds reasonable. I noticed the same thing but decided proceed in the same way as in `FreeC.Backend.Coq.Syntax`.Another advantage is that the code becomes more readable, because the smart constructors usually just provide lots of default arguments, which all had to be prefixed.